### ISSUE TITLE
go examples do not compile

### ISSUE THREAD

**Issue by insasho** [grpc/grpc-common#170]
https://github.com/grpc/grpc-common/blob/master/go/README.md suggests that `go get -u github.com/grpc/grpc-common/go/greeter_client` should compile cleanly. It does not.

Steps to repeat:
```
$ cat Dockerfile.grpc-go
FROM golang:1.4
RUN echo GOPATH=${GOPATH}
RUN echo PATH=${PATH}
RUN go get -u github.com/grpc/grpc-common/go/greeter_client
$ docker build -f Dockerfile.grpc-go .
Sending build context to Docker daemon 3.584 kB
Sending build context to Docker daemon 
Step 0 : FROM golang:1.4
 ---> aac88aa7c9ab
Step 1 : RUN echo GOPATH=${GOPATH}
 ---> Running in fe237133de48
GOPATH=/go
 ---> 9c6e22e0ca43
Removing intermediate container fe237133de48
Step 2 : RUN echo PATH=${PATH}
 ---> Running in 7d5ef3333231
PATH=/go/bin:/usr/src/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 ---> 93e47ee870b3
Removing intermediate container 7d5ef3333231
Step 3 : RUN go get -u github.com/grpc/grpc-common/go/greeter_client
 ---> Running in f305fb25f4df
# github.com/grpc/grpc-common/go/helloworld
src/github.com/grpc/grpc-common/go/helloworld/helloworld.pb.go:105: cannot use _Greeter_SayHello_Handler (type func(interface {}, context.Context, []byte) (proto.Message, error)) as type grpc.methodHandler in field value
```

Results: The package fails to compile with the following error:
```
# github.com/grpc/grpc-common/go/helloworld
src/github.com/grpc/grpc-common/go/helloworld/helloworld.pb.go:105: cannot use _Greeter_SayHello_Handler (type func(interface {}, context.Context, []byte) (proto.Message, error)) as type grpc.methodHandler in field value
```

Expected: The package should build.

**Comment by iamqizhao** [comment]
We recently updated the codegen plugin. But this is the one we left. will
update it shortly. Thanks for letting us know.

**Comment by iamqizhao** [comment]
fixed by #171
